pulp_ansible/app/models.py

```diff
@@ -259,6 +257,20 @@ class Meta:
         permissions = (("modify_ansible_repo_content", "Can modify ansible repository content"),)
 
 
+class AnsibleRepositoryCollection(BaseModel):
```

Comment:
You've basically done exactly what I was describing. The relating of a Collection to an `AnsibleRepository` seems right, but the name would be more appropriate if it was maybe `AnsibleCollectionDeprecated`. The presence of this record affects *only* the deprecated status.

### Thought experiment (this may be a terrible idea)

What happens if we get another piece of repo-level metadata, we'd end up creating another model type, and then another, etc. To avoid that architecture maybe we want to [use this pattern](https://docs.djangoproject.com/en/dev/topics/db/models/#extra-fields-on-many-to-many-relationships)

1) Leave the name as-is
2) Add the `deprecated` value defaulting to False

The major downside of this pattern is that you'll end up creating these relationships for every collection in every repo which to store an extremely sparse number of `deprecated=True` could be a bad idea performance wise.

I wanted to write this here, but likely we should just rename this model to `AnsibleCollectionDeprecated`. What do you think?

Reply:
I think we can rename it for now, but in the future, if we need another repo-level metadata I would prefer to rename this table and add the deprecated field